lib/internal/Magento/Framework/Stdlib/DateTime/Filter/DateTime.php

```diff
@@ -42,7 +42,7 @@ public function __construct(\Magento\Framework\Stdlib\DateTime\TimezoneInterface
     public function filter($value)
     {
         try {
-            $dateTime = $this->_localeDate->date($value, null, false);
```

Comment:
Can you please explain your fix, if I understand correctly, you pass `fasle` as the last parameter and `$this->_localeDate->date` will return date without the time, but on the next string we do `format('Y-m-d H:i:s')` that will add time.

Reply:
@VladimirZaets yes indeed. And this change happens because the issue is that Magento is trying to add the time in a format of the date that is probably not expected. See here `lib/internal/Magento/Framework/Stdlib/DateTime/Timezone.php:358`. That is always adding time as midnight. Maybe my solution is wrong but if you know a case that this addition is important I could definetely test it and figure it out. But from what I tested till now (saving all date attributes in the product object), time is definitely not important.